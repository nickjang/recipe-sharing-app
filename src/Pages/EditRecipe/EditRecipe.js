import React, { Component } from 'react';
import RecipeForm from '../../Components/RecipeForm/RecipeForm';
import UserService from '../../services/user-service';
import './EditRecipe.css';

class EditRecipe extends Component {
  handleEditSuccess = (recipeId) => {
    this.props.history.push(`/recipes/${recipeId}`);
  }

  handleDeleteSuccess = () => {
    const userId = UserService.getUserId();
    if (userId == null) return;
    this.props.history.push(`/users/${userId}/recipes`)
  }

  render() {
    const userId = UserService.getUserId();

    return (
      <>
        {userId === 1 &&
          <p className='note text-center mb-0'>Cannot create, edit, or delete recipes with demo account</p>}
        <RecipeForm
          onSuccess={this.handleEditSuccess}
          onDelete={this.handleDeleteSuccess}
          title='Edit Recipe'
          form='edit-recipe-form'
          submitButtonText='Make changes'
          fetchRecipe={true}
          recipeId={Number(this.props.match.params.recipeId)} />
      </>
    );
  }
}

EditRecipe.defaultProps = {
  match: {
    params: {
      recipeId: -1
    }
  },
  history: {
    push: () => { }
  }
}

export default EditRecipe;