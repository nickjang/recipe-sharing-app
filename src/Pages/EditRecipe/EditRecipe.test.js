import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditRecipe from './EditRecipe';

describe(`EditRecipe component`, () => {
  it('renders EditRecipe by default', () => {
    const wrapper = shallow(<EditRecipe />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})