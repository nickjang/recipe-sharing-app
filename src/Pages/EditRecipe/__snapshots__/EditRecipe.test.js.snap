// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipe component renders EditRecipe by default 1`] = `
<Fragment>
  <withRouter(RecipeForm)
    fetchRecipe={true}
    form="edit-recipe-form"
    onDelete={[Function]}
    onSuccess={[Function]}
    recipeId={-1}
    submitButtonText="Make changes"
    title="Edit Recipe"
  />
</Fragment>
`;
